### PR TITLE
Compact based on TTL

### PR DESCRIPTION
**Pull request by adam-yorwerth**

### CODE REVIEW

File: pipe-storage-postgresql/src/integration/groovy/com/tesco/aqueduct/pipe/storage/PostgresqlStorageIntegrationSpec.groovy [hunk 1]
```diff
@@ -202,147 +204,84 @@ class PostgresqlStorageIntegrationSpec extends StorageSpec {
...
         MessageResults result = storage.read(null, 0, ["cluster1"])
         List<Message> retrievedMessages = result.messages
 
-        then: 'duplicate messages arent deleted'
-        retrievedMessages.size() == 2
-
-        and: 'the correct compacted message list is returned in the message results'
-        result.messages*.offset*.intValue() == [1, 2]
-        result.messages*.key == ["A", "A"]
+        then: 'no messages are compacted'
```
Comment (sebastianlavender) [inline pipe-storage-postgresql/src/integration/groovy/com/tesco/aqueduct/pipe/storage/PostgresqlStorageIntegrationSpec.groovy:243]: This should be 'all messages are compacted

File: pipe-storage-postgresql/src/integration/groovy/com/tesco/aqueduct/pipe/storage/PostgresqlStorageIntegrationSpec.groovy [hunk 2]
```diff
@@ -202,147 +204,84 @@ class PostgresqlStorageIntegrationSpec extends StorageSpec {
...
+        and: 'all messages are read'
+        MessageResults result = storage.read(null, 0, ["cluster1"])
         List<Message> retrievedMessages = result.messages
 
-        then:
-        retrievedMessages.size() == 2
-        and: 'the correct compacted message list is returned in the message results'
-        result.messages*.offset*.intValue() == [1, 2]
-        result.messages*.key == ["A", "A"]
+        then: 'no messages are compacted'
```
Comment (sebastianlavender) [inline pipe-storage-postgresql/src/integration/groovy/com/tesco/aqueduct/pipe/storage/PostgresqlStorageIntegrationSpec.groovy:263]: As above naming not right

File: pipe-storage-postgresql/src/integration/groovy/com/tesco/aqueduct/pipe/storage/PostgresqlStorageIntegrationSpec.groovy [hunk 3]
```diff
@@ -202,147 +204,84 @@ class PostgresqlStorageIntegrationSpec extends StorageSpec {
...
+        retrievedMessages.size() == 1
     }
 
-    def 'Duplicate messages are not compacted when published after the threshold'() {
+    def 'Messages with null TTL arent compacted'() {
         given: 'a stored cluster'
         Long cluster1 = insertCluster("cluster1")
 
-        and: 'an existing data store with duplicate messages for the same key'
+        and: 'messages stored with cluster and TTL set to today'
```
Comment (ashish-sharma09) [inline pipe-storage-postgresql/src/integration/groovy/com/tesco/aqueduct/pipe/storage/PostgresqlStorageIntegrationSpec.groovy:271]: The ttl is null and not today for this test right?

**Review by ashish-sharma09 — APPROVED** [review]
LGTM

**Pull request merged**